PDBminer

```diff
@@ -73,30 +73,94 @@ parser.add_argument("-f", "--snakefile",
+
+    Parameters
+    ----------
+    df : The input dataframe generated from the input CSV.
 
+    Returns
+    -------
+    problems : If the uniprot accession number is not a string, the format
+    is wrong and will return a list of string(s) with the offending input.
+
+    """
     problems = []
     
-    for i in range(len(df)):    
+    #Uniprot Accession number check:
+    problematic_uniprots = df.uniprot[~df.uniprot.apply(lambda x: isinstance(x, str))]
+    for uniprot_value in problematic_uniprots:
+        problems.append(f"there is a problem with the uniprot input: {uniprot_value}")
+    
+    # If there are issues with the uniprot input, the program should end, 
+    # since this is the only mandatory element of the input.
+    if problems != []:
+        return problems
+    
+def retrieve_hugo_name(uniprot_accession_number):
+    """
+    This functon takes a uniprot accession number and retrieves a gene name
+    from the uniprot database. If Uniprot DB is down, the program will
+    end with an error.
+
+    Parameters
+    ----------
+    uniprot_accession_number : Accession number. 
+
+    Returns
+    -------
+    hugo_name : The listed gene name for the uniprot accession number.
+
+    """
+    
+    #try to get the hugoname from the uniprot database. 
+    # If the uniprot datbase is down, the program will exit.
+    try: 
+        response = requests.get(f"https://www.uniprot.org/uniprot/{uniprot_accession_number}.json")
+    except ConnectionError as e:
+        with open("log.txt", "a") as textfile:
+            textfile.write(f"EXITING: Uniprot database API controlled and rejected for {uniprot_accession_number} when finding the Hugo Name, connection error.\n")
+        exit(1)
+    
+    # If the response is successfull, hugo name is collected.
+    if response.status_code == 200:
+        uniprot_file_content = json.loads(response.text)
+        hugo_name = uniprot_file_content['genes'][0]['geneName']['value']
+    
+    # Otherwise the field is left empty
+    else: 
+        hugo_name = "NA"
+        with open("log.txt", "a") as textfile:
+            textfile.write(f"WARNING: No Hugo Name could be retrived with UniProt for {uniprot_id}. Gene name assigned NA.\n")
+        
+    return hugo_name
     
-        if type(df.uniprot[i]) != str:
-            problems.append(f"there is a problem with the uniprot input on the line of {df.loc[i]['uniprot']}")
     
-        if ('hugo_name' in df.columns) == False:
-            uniprot_url = f"https://www.uniprot.org/uniprot/{df.uniprot[i]}.json"
-            response = requests.get(uniprot_url)
-            uniprot_file_content = json.loads(response.text)
-            df.loc[i,"hugo_name"] = uniprot_file_content['genes'][0]['geneName']['value']
+def check_input_file(df):
+
+    problems = []
+    
+    # The rest of the columns are optional and may be populated in this function.
+    for i in range(len(df)):        
```

Comment:
iterating on a data frame, when it is necessary, it's better done by using `pd.DataFrame.iterrows()` rather than an explicit for loop